hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/conf/ConfigurationUpdateAssembler.java

```diff
@@ -76,19 +76,20 @@ private static void removeQueue(
       throw new IOException("Queue " + queueToRemove + " not found");
     }
     siblingQueues.remove(queueName);
-    String parentQueuePath = queueToRemove.substring(0, queueToRemove
+    String parentPath = queueToRemove.substring(0, queueToRemove
             .lastIndexOf('.'));
+    QueuePath parentQueuePath = new QueuePath(parentPath);
     proposedConf.setQueues(parentQueuePath, siblingQueues.toArray(
             new String[0]));
     String queuesConfig = CapacitySchedulerConfiguration.PREFIX
-            + parentQueuePath + CapacitySchedulerConfiguration.DOT
+            + parentPath + CapacitySchedulerConfiguration.DOT
             + CapacitySchedulerConfiguration.QUEUES;
     if (siblingQueues.isEmpty()) {
       confUpdate.put(queuesConfig, null);
       // Unset Ordering Policy of Leaf Queue converted from
       // Parent Queue after removeQueue
       String queueOrderingPolicy = CapacitySchedulerConfiguration.PREFIX
-              + parentQueuePath + CapacitySchedulerConfiguration.DOT
+              + parentPath + CapacitySchedulerConfiguration.DOT
               + ORDERING_POLICY;
       proposedConf.unset(queueOrderingPolicy);
       confUpdate.put(queueOrderingPolicy, null);
```

Comment:
Can we move all queue string operations from addQueue / removeQueue methods to the QueuePath object?

Reply:
Sure, actually most of them were already present in the QueuePath class, like getting the parent or the leaf part of a queue path. I refactored this class to use QueuePath here as well, to eliminate the string operations.